### PR TITLE
Update to vertx 3.9.0

### PR DESCRIPTION
**Pull request by cescoffier**
* Update Vert.x version to 3.9.0
* Update Netty version to 4.1.48.Final (aligned with the version used by Vert.x)
* Update Axle / Mutiny Vert.x client to 0.0.15 (version using Vert.x 3.9.0)
* Fix code following a breaking change in Vert.x 3.9 (Vert.x SQL Client)
* Update documentation to reflect this breaking change and document it

The quickstarts will also need to be updated.

### CODE REVIEW

**Comment by gsmet** [comment]
@cescoffier is this something you want in 1.4 or is it 1.5 material?

**Comment by gsmet** [comment]
(I would favor 1.5 except if we have very strong reasons to want it in 1.4)

**Comment by machi1990** [comment]
@cescoffier thanks for this update. I can revive my [redis-extension](https://github.com/machi1990/quarkus/tree/feat/vertx-redis-extension) work once this is merged.

**Comment by cescoffier** [comment]
@gsmet 1.5 is the target. I knew I will hit a few blockers. Already worked around a few of them, but now hitting a Vert.x issue that will need some fix upstream.

**Comment by cescoffier** [comment]
@vietj 

On Windows, I'm hitting a new behavior in the FileResolver:

```
2020-04-11T08:12:02.1127183Z 2020-04-11 08:12:02,096 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (vert.x-eventloop-thread-3) HTTP Request to /abc/not_found failed, error id: 8d6f8a61-e0f1-4141-9fbe-0b54e7e5cc50-1: java.nio.file.InvalidPathException: Illegal char <:> at index 120: D:\a\quarkus\quarkus\extensions\resteasy\deployment\target\vertx-cache\file-cache-93439211-4463-4880-bc3e-6fd8101e95a4\D:\a\quarkus\quarkus\extensions\resteasy\deployment\target\quarkus-dev-mode-test9360317579511630399\src\main\resources\META-INF\resources\index.html
2020-04-11T08:12:02.1131329Z 	at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
2020-04-11T08:12:02.1131593Z 	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
2020-04-11T08:12:02.1131836Z 	at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
2020-04-11T08:12:02.1132039Z 	at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
2020-04-11T08:12:02.1132255Z 	at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:229)
2020-04-11T08:12:02.1132461Z 	at java.base/java.io.File.toPath(File.java:2292)
2020-04-11T08:12:02.1132669Z 	at io.vertx.core.file.impl.FileResolver.unpackFromFileURL(FileResolver.java:178)
2020-04-11T08:12:02.1132881Z 	at io.vertx.core.file.impl.FileResolver.unpackFromFileURL(FileResolver.java:192)
2020-04-11T08:12:02.1133089Z 	at io.vertx.core.file.impl.FileResolver.unpackUrlResource(FileResolver.java:156)
2020-04-11T08:12:02.1133273Z 	at io.vertx.core.file.impl.FileResolver.resolveFile(FileResolver.java:139)
2020-04-11T08:12:02.1133487Z 	at io.vertx.core.impl.VertxImpl.resolveFile(VertxImpl.java:805)
2020-04-11T08:12:02.1133697Z 	at io.vertx.core.file.impl.FileSystemImpl$20.perform(FileSystemImpl.java:934)
2020-04-11T08:12:02.1133913Z 	at io.vertx.core.file.impl.FileSystemImpl$20.perform(FileSystemImpl.java:932)
2020-04-11T08:12:02.1134139Z 	at io.vertx.core.file.impl.FileSystemImpl$BlockingAction.handle(FileSystemImpl.java:975)
2020-04-11T08:12:02.1134357Z 	at io.vertx.core.file.impl.FileSystemImpl$BlockingAction.handle(FileSystemImpl.java:955)
2020-04-11T08:12:02.1134547Z 	at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$2(ContextImpl.java:316)
2020-04-11T08:12:02.1134754Z 	at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
2020-04-11T08:12:02.1134972Z 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
2020-04-11T08:12:02.1135194Z 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
2020-04-11T08:12:02.1135409Z 	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
2020-04-11T08:12:02.1135613Z 	at java.base/java.lang.Thread.run(Thread.java:834)
```

Look at the duplicated path. It seems that there is a "isAbsolute" call that get messed up.

**Comment by stuartwdouglas** [comment]
@cescoffier I have rebased and added some commits, lets see what CI says.

**Review by cescoffier — COMMENTED** [review]
@stuartwdouglas Thanks, your change looks good to me.

**Comment by gastaldi** [comment]
@cescoffier looks like this needs to be rebased again :stuck_out_tongue:

**Comment by cescoffier** [comment]
Rebase done.

**Comment by cescoffier** [comment]
But obviously something else changed...

**Comment by stuartwdouglas** [comment]
@cescoffier looks like you wiped out my commits?
https://github.com/quarkusio/quarkus/commit/8c0ae303b342b287a60de9581660d431b8bbb56e
https://github.com/quarkusio/quarkus/commit/6a04d88d9e8d7c39ee2d63d509fc9c7a818129ee

**Comment by cescoffier** [comment]
That is weird....

**Comment by cescoffier** [comment]
Didn't pull before rebasing... My bad.

**Comment by cescoffier** [comment]
@stuartwdouglas Fixed! Sorry.
I already covered the Cookie one, so I just-reapply the classloader fix.

**Comment by cescoffier** [comment]
@stuartwdouglas It failed on Windows with: [ERROR]   ClassLoadingInterruptTestCase.testClassLoaderWhenThreadInterrupted:28 expected: not equal but was: <class io.quarkus.bootstrap.classloader.InterruptClass>

**Comment by stuartwdouglas** [comment]
Oops, I linked the wrong version of that CL commit: https://github.com/quarkusio/quarkus/commit/fdace9cacaedbff19b5a1f1b71efdea771e5cacd

**Comment by stuartwdouglas** [comment]
@cescoffier I have force pushed

**Comment by cescoffier** [comment]
Thanks @stuartwdouglas !

File: bom/runtime/pom.xml [hunk 1]
```diff
@@ -2446,6 +2446,12 @@
                 <artifactId>smallrye-mutiny-vertx-pg-client</artifactId>
                 <version>${mutiny-client.version}</version>
             </dependency>
+            <dependency>
+                <!-- Used by the Reactive PG Client -->
+                <groupId>com.ongres.scram</groupId>
+                <artifactId>client</artifactId>
+                <version>2.1</version>
```
Comment (gsmet) [inline bom/runtime/pom.xml:2453]: The version should be parameterized.

File: docs/src/main/asciidoc/reactive-sql-clients.adoc [hunk 2]
```diff
@@ -184,14 +184,21 @@ It returns a `Uni` and thus can be composed to execute queries sequentially:
...
+    .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Orange')").execute())
+    .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Pear')").execute())
+    .flatMap(r -> client.query("INSERT INTO fruits (name) VALUES ('Apple')").execute())
     .await().indefinitely();
 ----
 
+[IMPORTANT]
+.Breaking Change
+====
+Vert.x 3.9 introduces a breaking change regarding the `query` methods.
```
Comment (gsmet) [inline docs/src/main/asciidoc/reactive-sql-clients.adoc:198]: We also need the Quarkus version here. People won't know in which version we introduced the change.

**Review by gsmet — CHANGES_REQUESTED** [review]
I added 2 minor comments.

There's one thing bothering me a bit: I don't like when we have commits making the project in a state where it doesn't compile at all.

When things don't compile, everything should be done in an atomic commit.

YMMV so feel free to ignore that comment.

CI is green so we should be able to merge that very soon.

File: docs/src/main/asciidoc/reactive-sql-clients.adoc (hunk 2 above)
Comment (cescoffier) [reply docs/src/main/asciidoc/reactive-sql-clients.adoc:198]: Yes, but I didn't know the version until today.

File: bom/runtime/pom.xml (hunk 1 above)
Comment (cescoffier) [reply bom/runtime/pom.xml:2453]: oops missed it.

File: docs/src/main/asciidoc/reactive-sql-clients.adoc (hunk 2 above)
Comment (cescoffier) [reply docs/src/main/asciidoc/reactive-sql-clients.adoc:198]: Note that I need to add a few lines to the breaking changes document

Comment (cescoffier) [reply docs/src/main/asciidoc/reactive-sql-clients.adoc]: Done.

**Comment by cescoffier** [comment]
CI issue unrelated (Maven tooling)

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Merged, thanks!

**Pull request merged**